core/src/main/java/org/infinispan/stream/impl/local/EntryStreamSupplier.java

```diff
@@ -44,7 +45,9 @@ public EntryStreamSupplier(Cache<K, V> cache, ConsistentHash hash, Supplier<Stre
       }
       if (segmentsToFilter != null && hash != null) {
          log.tracef("Applying segment filter %s", segmentsToFilter);
-         stream = stream.filter(k -> segmentsToFilter.contains(hash.getSegment(k.getKey())));
+         BitSet bitSet = new BitSet(hash.getNumSegments());
```

Comment:
Hrmm unfortunately it requires changing quite a few classes cause I want to have access to the hash to properly initialize the BitSet to the correct size and avoid any resizings.